exec/java-exec/src/test/java/org/apache/drill/BaseTestQuery.java

```diff
@@ -593,4 +610,49 @@ private void convert(List<QueryDataBatch> batches) throws SchemaChangeException
       }
     }
   }
+
+  private static String replaceWorkingPathInString(String orig) {
+    return orig.replaceAll(Pattern.quote("[WORKING_PATH]"), Matcher.quoteReplacement(TestTools.getWorkingPath()));
+  }
+
+  protected static void copyDirectoryIntoTempSpace(String resourcesDir) throws IOException {
+    copyDirectoryIntoTempSpace(resourcesDir, null);
+  }
+
+  protected static void copyDirectoryIntoTempSpace(String resourcesDir, String destinationSubDir) throws IOException {
+    Path destination = destinationSubDir != null ? new Path(getDfsTestTmpSchemaLocation(), destinationSubDir)
+        : new Path(getDfsTestTmpSchemaLocation());
+    fs.copyFromLocalFile(
+        new Path(replaceWorkingPathInString(resourcesDir)),
+        destination);
+  }
+
+  /**
+   * Metadata cache files include full paths to the files that have been scanned.
```

Comment:
For older files with the marker, should we just replace the marker to be relative and take advantage of this improvement? Can that be done without having to edit the old files?